site/docs/tools/iceberg/spark.md

```diff
@@ -8,6 +8,8 @@ To access Nessie on Iceberg from a spark cluster make sure the `spark.jars` spar
 | | `iceberg-spark-runtime` *(required)* | `nessie-spark-extensions` *(optional)* |
 |---|:---:|:---:|
 {%- for (sparkver, scalaver) in [
+  ('3.4', '2.12'),
+  ('3.4', '2.13'),
   ('3.3', '2.12'),
   ('3.3', '2.13'),
   ('3.2', '2.12'),
```

Comment:
I think some more places need to be handled. 

1. CI job for spark-3.4
https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/.github/workflows/ci.yml#L721

2. GC Integration tests can be updated to use latest spark version 
https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/gc/gc-tool-inttest/build.gradle.kts#L28

https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/gc/gc-iceberg-inttest/build.gradle.kts#L26

3. Jackson dependency for spark-3.4
https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/integrations/spark-extensions/build.gradle.kts#L43

4. Need to add 3.4 here 
https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/site/docs/tools/sql.md?plain=1#L6

5. Maybe defaults can be changed to spark-3.4
https://github.com/projectnessie/nessie/blob/e406df9c2d1a02fb77f5520c9eeda161925e39c7/site/macros.py#L30-L41

Reply:
The goal of this PR is _not_ to change defaults that work fine in every place w/o a pressing need.